addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -3014,7 +3014,35 @@ export class Wysiwyg extends Component {
                 }
             }
         }
-        const newAttachmentSrc = await this._serviceRpc(
+        // The field 'description' of the attachment is used to store
+        // data such as the shape, the shape color, the filter, the
+        // width and the quality. This is necessary in order to be able
+        // to display the correct parameters at the re-edition.
+        let description = "";
+        for (const imgAttribute of weUtils.IMAGE_ATTRIBUTES_STORED_IN_ATTACHMENT) {
+            if (el.dataset[imgAttribute]) {
+                description = description.concat(`${imgAttribute}:${el.dataset[imgAttribute]} `);
+            }
+        }
+        let classesToSave = "";
+        for (const classAttribute of weUtils.CLASSES_STORED_IN_ATTACHMENT) {
+            if (el.classList.contains(classAttribute)) {
+                if (!classesToSave) {
+                    classesToSave = `classes:${classAttribute}`;
+                } else {
+                    classesToSave = classesToSave.concat(`,${classAttribute}`);
+                }
+            }
+        }
+        description = description.concat(`${classesToSave}`);
+        if (description) {
+            // Stored the description inside "| " and " |". Those chains of
+            // characters are only used as marker to spot the wanted description
+            // data more easily inside the description field of the attachment.
+            description = "| ".concat(description).concat(" |");
+        }
+        el.dataset.oeTmpEmbedded = description;
```

Comment:
At this stage, we know that the image has been modified. If `description` is empty, the `oe-tmp-embedded` data attribute should still be added as it could mean that the user removed all the options (shape, filter, etc...) previously set. Thanks to the empty `oe-tmp-embedded`, the data related options will be removed from the db.